cli/src/commands/publish.ts

```diff
@@ -0,0 +1,233 @@
+import { Command } from "commander";
+import chalk from "chalk";
+import { select, confirm, password } from "../prompts.js";
+import {
+  createSpinner,
+  printSuccess,
+  printError,
+  printNodeInfo,
+  symbols,
+  maskString,
+} from "../ui.js";
+import { getApiKey, getEnvConfig, getPrivateKey, setPrivateKey } from "../config.js";
+import {
+  getDraftNode,
+  listNodes,
+  publishNode,
+  signerFromPkey,
+} from "@desci-labs/nodes-lib/node";
+
+export function createPublishCommand(): Command {
+  return new Command("publish")
+    .description("Publish a node to Codex (requires signing)")
+    .argument("[node]", "Node UUID or partial UUID")
+    .option("-k, --private-key", "Prompt for private key (won't be saved)")
+    .option("--save-key", "Save the private key for future use")
+    .option("--mint-doi", "Request DOI minting during publish")
+    .action(async (nodeArg: string | undefined, options) => {
+      try {
+        // Check API key
+        if (!getApiKey()) {
+          printError(
+            "No API key configured. Run: nodes-cli config login",
+          );
+          process.exit(1);
+        }
+
+        let targetUuid = nodeArg;
+
+        // If no node specified, show picker
+        if (!targetUuid) {
+          const spinner = createSpinner("Fetching your nodes...");
+          spinner.start();
+
+          const { nodes } = await listNodes();
```

Comment:
Perhaps it would make sense to filter only nodes that have drafts (nodes with unpublished edits) here, idk how easy it is to find those though.